src/cmd-push-container-manifest

```diff
@@ -60,17 +59,19 @@ def main():
             images.append(f"oci-archive:{ociarchive}")
 
         # Create/Upload the manifest list to the container registry
-        with tempfile.NamedTemporaryFile() as digestfile:
-            create_and_push_container_manifest(
-                args.repo, args.tags, images, args.v2s2, digestfile.name)
-            digestfile.seek(0)
-            digest = digestfile.read().decode('utf-8').strip()
+        manifest = create_and_push_container_manifest(
+            args.repo, args.tags, images, args.v2s2)
 
-        # Update the meta.json in each build/arch metadata
-        for _, buildmeta in buildmetas.items():
-            buildmeta[args.metajsonname] = {
+        assert len(manifest['manifests']) == len(buildmetas)
+        for manifest in manifest['manifests']:
+            arch = manifest['platform']['architecture']
+            if arch == 'arm64':
+                arch = 'aarch64'
+            elif arch == 'amd64':
+                arch = 'x86_64'
+            buildmetas[arch][args.metajsonname] = {
                 'image': args.repo,
-                'digest': digest
+                'digest': manifest['digest']
             }
             buildmeta.write(artifact_name=args.metajsonname)
```

Comment:
bug - needs to be updated now?

```diff
- buildmeta.write(artifact_name=args.metajsonname)
+ buildmetas[arch].write(artifact_name=args.metajsonname)
```

Reply:
Good catch! This worked in local tests because I was only uploading `x86_64` and the `buildmeta` declared higher up when we gather the OCI archives was still in scope.